src/Config.php

```diff
@@ -69,4 +51,79 @@ public function __construct($config)
      * @var array
      */
     public $options;
+
+    /**
+     * Create a new SQL connection configuration from the given configuration key-value pairs
+     *
+     * The given configuration is constraint to the class public properties.
+     * Keys will be converted to camelCase, e.g. use_ssl → useSsl.
+     *
+     * @param iterable $config Configuration key-value pairs
+     *
+     * @throws InvalidArgumentException If $config is not iterable
+     */
+    public function __construct($config)
+    {
+        if (! is_iterable($config)) {
+            throw new InvalidArgumentException(sprintf(
+                '%s expects parameter one to be iterable, got %s instead',
+                __METHOD__,
+                get_php_type($config)
+            ));
+        }
+
+        foreach ($config as $key => $value) {
+            $this->setProperty(Str::camel($key), $value);
+        }
+    }
+
+    public function props(): \Generator
+    {
+        foreach (get_object_vars($this) as $name => $ignored) {
+            yield $name;
+        }
+    }
+
+    /**
+     * @param string $name
+     *
+     * @return string|null
+     */
+    public function __get(string $name)
+    {
+        foreach ($this->props() as $key) {
+            if ($name === $key) {
+                return $this->$name;
+            }
+        }
+
+        if (array_key_exists($name, $this->extraSettings)) {
+            return $this->extraSettings[$name];
+        }
+
+        return null;
```

Comment:
I would throw an exception to protect against unexpected results:

`throw new OutOfRangeException(sprintf('Property %s does not exist', $name));`